legacy_code/src/device_parser.py

```diff
@@ -22,12 +22,14 @@ class DeviceParser(Parser):
         'time': str
     }
 
+    BROWSER_PATTERN = '"browser'
+
     def get_uuid(self, data):
         return data.get('id')
 
-    def format_check(self, line, line_num):
-        if ('event_properties' not in line) or ('browser' not in line) or (not self.has_valid_json(line, line_num)):
-            return True
+    def is_valid_format(self, line, line_num):
+        if self.BROWSER_PATTERN in line:
+            return super(DeviceParser, self).is_valid_format(line, line_num)
```

Comment:
Doesn't python3 use just `super()` with no args?

Reply:
what do you mean by `super()` with no args?

Reply:
In Python 3, I thought the normal pattern is to not pass any arguments when calling `super()`.

https://docs.python.org/3/library/functions.html#super

Reply:
I see. It is true, it works both ways.  Because I usually make my code PYTHON2 / 3 compatible, and this is more explicit about what you are doing. And if you insist, I can do `super()` with no problem.